apis/v1alpha2/validation/gateway.go

```diff
@@ -26,6 +27,19 @@ import (
 	gatewayv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
 )
 
+var (
+	exists          = struct{}{}
```

Comment:
`exists` seems a bit non-idiomatic for Go. I'd just inline the struct literals where needed.

Reply:
+1 this is not something we have elsewhere in the codebase, let's leave this out.